### ISSUE TITLE
ATS9: Crash on ProxySession::do_api_callout()

### ISSUE THREAD

**Issue by masaori335** [apache/trafficserver#6511]
```
[ 0  ] libpthread-2.17.so     waitpid                                                      ( :undefined               ) 
[ 1  ] traffic_server         crash_logger_invoke(int, siginfo_t*, void*)                  ( Crash.cc:168             ) 
[ 2  ] libpthread-2.17.so     0x7fc6cfa2f5f0                                               ( :undefined               ) 
[ 3  ]                       0x0                                                          ( :undefined               ) 
[ 4  ] traffic_server         ProxySession::do_api_callout(TSHttpHookID)                   ( ProxySession.cc:171      ) 
[ 5  ] traffic_server         Http1ClientSession::destroy()                                ( Http1ClientSession.cc:79 ) 
[ 6  ] traffic_server         HttpSM::kill_this()                                          ( HttpSM.cc:6946           ) 
[ 7  ] traffic_server         HttpSM::state_api_callback(int, void*)                       ( HttpSM.cc:1351           ) 
[ 8  ] traffic_server         TSHttpTxnReenable                                            ( InkAPI.cc:6096           ) 
[ 9  ] authproxy.so (deleted) 0x7fc692d66878                                               ( :undefined               ) 
[ 10 ] authproxy.so (deleted) 0x7fc692d65d15                                               ( :undefined               ) 
[ 11 ] traffic_server         INKContInternal::handle_event(int, void*)                    ( InkAPI.cc:1101           ) 
[ 12 ] traffic_server         PluginVC::process_read_side(bool)                            ( I_Continuation.h:190     ) 
[ 13 ] traffic_server         PluginVC::process_write_side(bool)                           ( PluginVC.cc:569          ) 
[ 14 ] traffic_server         PluginVC::main_handler(int, void*)                           ( PluginVC.cc:225          ) 
[ 15 ] traffic_server         EThread::process_event(Event*, int)                          ( I_Continuation.h:190     ) 
[ 16 ] traffic_server         EThread::process_queue(Queue<Event, Event::Link_link>*, i... ( UnixEThread.cc:176       ) 
[ 17 ] traffic_server         EThread::execute_regular()                                   ( UnixEThread.cc:236       ) 
[ 18 ] traffic_server         EThread::execute()                                           ( UnixEThread.cc:345       ) 
[ 19 ] traffic_server         spawn_thread_internal(void*)                                 ( Thread.cc:92             ) 
[ 20 ] libpthread-2.17.so     start_thread                                                 ( :undefined               )
```

**Comment by masaori335** [comment]
```
(gdb) frame 4
#4  0x000055dc92e8b912 in Http1ClientSession::free (this=0x7fc6178e50c0) at Http1ClientSession.cc:115
115	  this->do_io_write(nullptr, 0, nullptr);
```

**Comment by shinrich** [comment]
Sounds like the crashes we were seeing that should have been addressed by https://github.com/apache/trafficserver/pull/6469.  But I see that commit is already in the 9.0.x branch.

**Comment by zwoop** [comment]
Mark T. and I looked over our builds, and everything prior to this should be in the build that we see this crasher with:

```
commit 919b9fa6b3d7366cda181aab0172c32308945360
Author:     Brian Neradt <brian.neradt@gmail.com>
AuthorDate: Tue Mar 10 21:31:26 2020 -0500
Commit:     Leif Hedstrom <zwoop@apache.org>
CommitDate: Tue Mar 10 20:31:41 2020 -0600
```

This includes the fixes in #6469 ...

**Comment by masaori335** [comment]
Duplicated with #6527. Closing this because #6527 has more details (symbols, history...)
